python/cog/server/worker.py

```diff
@@ -533,17 +539,27 @@ async def _aloop(
         assert isinstance(self._events, LockedConnection)
         self._events = AsyncConnection(self._events.connection)
 
-        task = None
-
         async with asyncio.TaskGroup() as tg:
+            tasks = weakref.WeakValueDictionary[str | None, asyncio.Task[Any]]()
             while True:
                 e = cast(Envelope, await self._events.recv())
-                if isinstance(e.event, Cancel) and task and self._cancelable:
+                if isinstance(e.event, Cancel):
+                    if not self._cancelable:
```

Comment:
this isn't a useful flag for concurrent predictions (it's set and cleared in _handle_predict_error; but in the concurrent context it's set whenever *any* prediction starts and cleared whenever *any* prediction finishes); but I think we can rely entirely on whether the prediction is in the weak value dictionary.